gqcp/include/Domain/DiscreteDomain.hpp

```diff
@@ -120,7 +120,7 @@ class DiscreteDomain:
      *
      *  @return     The i-th element with values 0 or 1 depending on whether the i-th element is in the domain.
      */
-    bool operator()(const size_t i) const { return this->domain[i]; }
+    bool operator[](const size_t i) const { return this->domain[i]; }
```

Comment:
I chose the `()` method since it is in line with the style that `GQCP` currently uses in `GQCP::Matrix` classes and derivatives thereof.